File: examples/typescript/policies/updatePolicy.ts
// Usage: pnpm tsx policies/updatePolicy.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();
const policy = await cdp.policies.createPolicy({
  policy: {
    scope: "account",
    description: "Initial Allowlist Policy",
    rules: [
      {
        action: "accept",
        operation: "signEvmTransaction",
        criteria: [
          {
            type: "evmAddress",
            addresses: ["0x000000000000000000000000000000000000dEaD"],
            operator: "in",
          },
        ],
      },
    ],
  },
});

const updatedPolicy = await cdp.policies.updatePolicy({
  id: policy.id,
  policy: {
    description: "Updated Denylist Policy",
    rules: [
      {
        action: "accept",
        operation: "signEvmTransaction",
        criteria: [
          {
            type: "evmAddress",
            addresses: ["0x000000000000000000000000000000000000dEaD"],
            operator: "not in",
          },
        ],
      },
    ],
  },
});
console.log("Updated policy: ", JSON.stringify(updatedPolicy, null, 2));

File: examples/python/policies/update_policy.py
# Usage: uv run python policies/update_policy.py

import asyncio

from cdp import CdpClient
from dotenv import load_dotenv
from cdp.policies.types import CreatePolicyOptions, EvmAddressCriterion, SignEvmTransactionRule, UpdatePolicyOptions

load_dotenv()

async def main():
    async with CdpClient() as cdp:
        policy = CreatePolicyOptions(
            scope="account",
            description="Initial Allowlist Policy",
            rules=[
                SignEvmTransactionRule(
                    action="accept",
                    criteria=[
                        EvmAddressCriterion(
                            addresses=["0x000000000000000000000000000000000000dEaD"],
                            operator="in",
                        ),
                    ],
                )
            ],
        )

        result = await cdp.policies.create_policy(policy=policy)
        print("Created account policy: ", result.id)

        updated_policy = UpdatePolicyOptions(
            description="Updated Denylist Policy",
            rules=[
                SignEvmTransactionRule(
                    action="accept",
                    criteria=[
                        EvmAddressCriterion(
                            addresses=["0x000000000000000000000000000000000000dEaD"],
                            operator="not in",
                        ),
                    ],
                )
            ],
        )

        print("Updating account policy: ", result.id)
        updated_result = await cdp.policies.update_policy(id=result.id, policy=updated_policy)
        print("Updated account policy: ", updated_result)


asyncio.run(main())
